src/compose/ComposeBox.js

```diff
@@ -320,23 +324,22 @@ class ComposeBox extends PureComponent<Props, State> {
   };
 
   handleSend = () => {
-    const { dispatch, narrow, caughtUp, _ } = this.props;
+    const { dispatch, _ } = this.props;
     const { message } = this.state;
+    const narrow = this.getDestinationNarrow();
 
-    if (!caughtUp.newer) {
-      showErrorAlert(_('Failed to send message'));
-      return;
-    }
-
-    dispatch(addToOutbox(this.getDestinationNarrow(), message));
+    // It's safe to use getState directly here, since we aren't updating UI state from it.
+    // See CZO for more details:
+    // https://chat.zulip.org/#narrow/stream/48-mobile/topic/Failed.20to.20send.20on.20Android/near/1158150
+    if (this.props.onSend(store.getState, dispatch, _, message, narrow)) {
+      this.setMessageInputValue('');
 
-    this.setMessageInputValue('');
+      if (this.mentionWarnings.current) {
+        this.mentionWarnings.current.clearMentionWarnings();
+      }
 
-    if (this.mentionWarnings.current) {
-      this.mentionWarnings.current.clearMentionWarnings();
+      dispatch(sendTypingStop(narrow));
```

Comment:
Oh, also: looks like this commit starts having us use the destination narrow for the "Failed to send message" check, which is relevant to [a known bug](https://chat.zulip.org/#narrow/stream/48-mobile/topic/Failed.20to.20send.20on.20Android/near/1155882):

> Hmm, I'm getting a "failed to send message" popup when trying to message in #**test here** , and I'm not sure why.

[…]

> yeah, it bases it on whether you have any unfetched messages in the narrow you're in, not the topic you're trying to send to. that's just a bug no matter how you slice it.

So that's something else that would be great to stand on its own, and link to that conversation (I'm not finding a corresponding issue). 🙂 Hopefully extracting that out will leave this as a commit we can mark NFC (good for reviewing and history-reading), or not too many steps off from being NFC.

Reply:
Added a commit to just do the `const narrow = this.getDestinationNarrow()` change, and added a link to that CZO discussion in the commit.

Removing `state.topic` as you describe seems like a good plan to me.